### PR TITLE
added recipe for kcodecs

### PR DESCRIPTION
**Pull request by boussaffawalid**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

This recipe depends on:
* https://github.com/conan-io/conan-center-index/pull/3211
* https://github.com/conan-io/conan-center-index/pull/1759

### CODE REVIEW

File: recipes/kcodecs/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("extra-cmake-modules/5.75.0")
+
+    def requirements(self):
+        self.requires("gperf/3.1")
+        self.requires("qt/5.15.1")
```
Comment (gocarlos) [inline recipes/kcodecs/all/conanfile.py:31]: Not available yet: https://github.com/conan-io/conan-center-index/pull/1759

Comment (boussaffawalid) [reply recipes/kcodecs/all/conanfile.py:31]: Im preparation this until https://github.com/conan-io/conan-center-index/pull/3211 and https://github.com/conan-io/conan-center-index/pull/1759 get merged

File: recipes/kcodecs/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.5)
```
Comment (uilianries) [inline recipes/kcodecs/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

File: recipes/kcodecs/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+project(PackageTest CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(Qt5 CONFIG REQUIRED Core)
+
+add_executable(example example.cpp)
+target_link_libraries(example CONAN_PKG::kcodecs Qt5::Core)
```
Comment (uilianries) [inline recipes/kcodecs/all/test_package/CMakeLists.txt:10]:
```suggestion
target_link_libraries(example CONAN_PKG::kcodecs Qt5::Core)
set_property(TARGET example PROPERTY CXX_STANDARD 11)
```

**Pull request closed without merging**